### PR TITLE
ENH: add suffixes argument to compare

### LINKED ISSUE #44354: ENH: add suffixes argument to DataFrame.compare
#### Is your feature request related to a problem?

It would be convenient to choose other suffixes in `DataFrame.compare` than `self, other`

#### Describe the solution you'd like

Add argument `suffixes` so users can specify them.

#### API breaking implications

None

#### Additional context

```python
df = pd.DataFrame(
    {"col1": ["a", "b", "c"], "col2": [1.0, 2.0, np.nan], "col3": [1.0, 2.0, 3.0]},
    columns=["col1", "col2", "col3"],
)
df2 = df.copy()
df2.loc[0, "col1"] = "c"
df2.loc[2, "col3"] = np.nan
diff = df.compare(df2, suffixes=["left", "right"])
print(diff)

  col1       col3      
  left right left right
0    a     c  NaN   NaN
2  NaN   NaN  3.0   NaN
```

### PR DESCRIPTION
**Pull request by erfannariman**
- [x] closes #44354 
- [ ] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/frame/methods/test_compare.py [hunk 1]
```diff
@@ -180,3 +180,20 @@ def test_compare_unaligned_objects():
...
+        columns=["col1", "col2", "col3"],
+    )
+    df2 = df1.copy()
+    df2.loc[0, "col1"] = "c"
+    df2.loc[2, "col3"] = 4.0
+
+    suffixes = ["left", "right"]
+    comp = df1.compare(df2, suffixes=suffixes)
+
+    result_suffixes = comp.columns.get_level_values(1).unique()
```
Comment (jreback) [inline pandas/tests/frame/methods/test_compare.py:198]: can you use tm.assert_frame_equal here to assert the results

File: pandas/core/generic.py [hunk 2]
```diff
@@ -8532,7 +8533,7 @@ def compare(
         else:
             axis = self._get_axis_number(align_axis)
 
-        diff = concat([self, other], axis=axis, keys=keys)
+        diff = concat([self, other], axis=axis, keys=suffixes)
```
Comment (jreback) [inline pandas/core/generic.py:8536]: these must be a tuple and non-None right? can you add a check and tests the cases (e.g. this is similar to what we checkin merge for example)

File: pandas/core/frame.py [hunk 3]
```diff
@@ -7246,12 +7246,14 @@ def compare(
         align_axis: Axis = 1,
         keep_shape: bool = False,
         keep_equal: bool = False,
+        suffixes: Suffixes = ("self", "other"),
```
Comment (jreback) [inline pandas/core/frame.py:7249]: pls update doc string & add a versionadded.

if we have this in the docs anywhere else (i don't think so but pls check), maybe needs updating

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by jreback — CHANGES_REQUESTED** [review]
also add a note in other enhancements section for 1.4

File: pandas/core/frame.py (hunk 3 above)
Comment (jreback) [reply pandas/core/frame.py:7300]: can you update the doc-string, needs a versionadded 1.4. tag

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by mroeschke** [comment]
Thanks for the pull request, but this pull request appears to have gone stale. If interested in continuing, please merge in main and address the comments and this PR can be reopened. Closing in the meantime

**Pull request closed without merging**